src/tiled/flipmapobjects.cpp

```diff
@@ -64,15 +64,27 @@ FlipMapObjects::FlipMapObjects(MapDocument *mapDocument,
         } else { //computing bound rect for other
             boundaryObjectsRect = boundaryObjectsRect.united(objectTransform.mapRect(object->bounds()));
         }
+
+        mOldCellStates.append(object->propertyChanged(MapObject::CellProperty));
+        mNewCellStates.append(true);;
+
+        mOldRotationStates.append(object->propertyChanged(MapObject::RotationProperty));
+        mNewRotationStates.append(true);;
     }
     mObjectsCenter = boundaryObjectsRect.center();
 }
 
 void FlipMapObjects::flip()
 {
     //flip objects
-    for (MapObject *object : mMapObjects)
-        object->flip(mFlipDirection, mObjectsCenter);
+    for (int i = 0; i < mMapObjects.size(); ++i) {
+        mMapObjects[i]->flip(mFlipDirection, mObjectsCenter);
+
+        mMapObjects[i]->setPropertyChanged(MapObject::CellProperty, mNewCellStates[i]);
```

Comment:
I'm a little concerned about this. I think flipping some tile object is not actually expected to override the referenced tile, but just its flipped state. I think it's not very important to deal with this now, but it's something to keep in mind.